packages/@aws-cdk/aws-appsync/lib/graphqlapi.ts

```diff
@@ -249,6 +241,61 @@ export interface GraphQLApiProps {
 
 }
 
+/**
+ * A IamResource Class for IAM Permissions
+ */
+export class IamResource {
+  /**
+   * Generate the resourceNames given custom arns
+   *
+   * @param arns The custom arns that need to be permissioned
+   */
+  public static custom(...arns: string[]): IamResource {
+    if (arns.length === 0) {
+      throw new Error('Missing custom arn definition.');
+    }
+    return new IamResource(arns);
+  }
+
+  /**
+   * Generate the resourceNames given a type and fields
```

Comment:
same comment mentioned earlier applies re: doc strings for all the methods in this file